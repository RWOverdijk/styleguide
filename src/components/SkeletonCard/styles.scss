/** @define SkeletonCard */

@import "../../css/_global";

.Van-SkeletonCard {
  width: 363px;
  border: 2px solid $color-grey-bright;
  background-color: $color-white;
  padding: 20px;
}
