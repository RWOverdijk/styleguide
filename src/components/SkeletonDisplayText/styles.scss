/** @define SkeletonDisplayText */

@import "../../css/_global";
@import "../../css/mixins";

.Van-SkeletonDisplayText {
  @include skeleton-shimmer;
  @include skeleton-style;

  backface-visibility: hidden;
  max-width: 12rem;
  position: relative;

  &:not(:first-child) {
    margin-top: 1.2rem;
  }

  &:not(:last-child) {
    margin-bottom: 1.2rem;
  }

  &--small {
    height: 2.8rem;
  }

  &--medium {
    height: 3.2rem;
  }

  &--large {
    height: 4.4rem;
  }

  &--extraLarge {
    height: 6.4rem;
  }

  &--full {
    max-width: 100%;
    width: 100%;
  }
}
