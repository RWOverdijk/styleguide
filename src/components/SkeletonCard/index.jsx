import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

class SkeletonCard extends PureComponent {
  render() {
    const {
      className,
      children,
      ...props
    } = this.props;

    return (
      <div
        className={classnames('Van-SkeletonCard', className)}
        {...props}
      >
        {children}
      </div>
    );
  }
}

SkeletonCard.defaultProps = {
  className: '',
  children: '',
};

SkeletonCard.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]),
};

export default SkeletonCard;
