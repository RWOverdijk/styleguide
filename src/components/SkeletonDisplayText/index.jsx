import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

class SkeletonDisplayText extends PureComponent {
  render() {
    const {
      className,
      size,
      full,
      ...props
    } = this.props;
    return (
      <div
        className={classnames('Van-SkeletonDisplayText', {
          'Van-SkeletonDisplayText--small': size === 'small',
          'Van-SkeletonDisplayText--medium': size === 'medium',
          'Van-SkeletonDisplayText--large': size === 'large',
          'Van-SkeletonDisplayText--extraLarge': size === 'extraLarge',
          'Van-SkeletonDisplayText--full': full,
        }, className)}
        {...props}
      />
    );
  }
}

SkeletonDisplayText.defaultProps = {
  className: '',
  size: 'medium',
  full: false,
};

SkeletonDisplayText.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large',
    'extraLarge',
  ]),
  full: PropTypes.bool,
};

export default SkeletonDisplayText;
