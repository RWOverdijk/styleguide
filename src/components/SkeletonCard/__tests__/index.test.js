import React from 'react';
import renderer from 'react-test-renderer';
import SkeletonDisplayText from '../../SkeletonDisplayText';
import SkeletonCard from '../';

describe("<SkeletonCard>", () => {
  it("renders with a string as children", () => {
    const tree = renderer.create(
      <SkeletonCard>My card</SkeletonCard>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with a component as children", () => {
    const tree = renderer.create(
      <SkeletonCard>
        <SkeletonDisplayText />
      </SkeletonCard>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <SkeletonCard className="my-class">My Col</SkeletonCard>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <SkeletonCard data-id="0">My Col</SkeletonCard>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
