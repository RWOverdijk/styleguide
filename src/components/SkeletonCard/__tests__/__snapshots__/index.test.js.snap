// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonCard> renders with a component as children 1`] = `
<div
  className="Van-SkeletonCard"
>
  <div
    className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium"
  />
</div>
`;

exports[`<SkeletonCard> renders with a string as children 1`] = `
<div
  className="Van-SkeletonCard"
>
  My card
</div>
`;

exports[`<SkeletonCard> renders with specific attributes 1`] = `
<div
  className="Van-SkeletonCard"
  data-id="0"
>
  My Col
</div>
`;

exports[`<SkeletonCard> renders with specific className 1`] = `
<div
  className="Van-SkeletonCard my-class"
>
  My Col
</div>
`;
